rnd/autogpt_server/schema.prisma

```diff
@@ -212,6 +214,50 @@ model AgentGraphExecutionSchedule {
   @@index([isEnabled])
 }
 
+model AnalyticsDetails {
+  // PK uses gen_random_uuid() to allow the db inserts to happen outside of prisma
+  // typical uuid() inserts are handled by prisma
+  id String @id @default(dbgenerated("gen_random_uuid()"))
+
+  createdAt DateTime @default(now())
+  updatedAt DateTime @default(now()) @updatedAt
+
+  // Link to User model
+  userId String
+  user   User   @relation(fields: [userId], references: [id])
+
+  // Analytics Categorical data used for filtering (indexable w and w/o userId)
+  type String
+
+  // Analytic Specific Data. We should use a union type here, but prisma doesn't support it.
+  data Json?
+
+  // Indexable field for any count based analytical measures like page order clicking, tutorial step completion, etc.
+  dataIndex String?
+
+  @@index([userId, type], name: "analyticsDetails")
+  @@index([type])
+}
+
+model AnalyticsMetrics {
+  id        String   @id @default(dbgenerated("gen_random_uuid()"))
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  // Analytics Categorical data used for filtering (indexable w and w/o userId)
+  analyticMetric String
+  // Any numeric data that should be counted upon, summed, or otherwise aggregated.
+  value          Float
+  // Any string data that should be used to identify the metric as distinct.
+  // ex: '/build' vs '/market'
+  dataString     String?
+
+  // Link to User model
+  userId String
+  user   User   @relation(fields: [userId], references: [id])
+
+  @@index(fields: [analyticMetric, userId, dataString], name: "analytics_metric_index")
```

Comment:
userId Index between the analyticMetric & dataString would be problematic, I think we need to remove it.
you won't be able to do where(metric=X, dataString=Y) quickly.

And userId indexing is by default added by prisma I think.

Reply:
When would you want to do this not associated with a user?

Reply:
tbf you don't need the indexing at all for the analytics since you will be querying this on different database + faster insert

Reply:
select count(*) where metric = event and datastring = event_data ?

Reply:
Works for me, I’ll drop it